src/main/java/seedu/duke/models/logic/Storage.java

```diff
@@ -0,0 +1,240 @@
+package seedu.duke.models.logic;
+
+import seedu.duke.models.schema.ModuleList;
+import seedu.duke.models.schema.Schedule;
+import seedu.duke.models.schema.Student;
+import seedu.duke.utils.exceptions.CorruptedFileException;
+import seedu.duke.utils.exceptions.MissingFileException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Storage {
+
+    private String userName;
+
+    private String userDirectory = System.getProperty("user.dir");
+
+    /**
+     * Constructs a new Storage instance with the specified file path.
+     *
+     * @param userName The path to the storage file.
+     */
+    public Storage(String userName) {
+        this.userName = userName;
+    }
+
+    /**
+     * Creates a "schedule.txt" file in the storage directory.
+     */
+    public void createUserStorageFile() {
+        String storageDirectory = userDirectory + "/" + userName;
+        createDirectory(storageDirectory);
+        createFileInDirectory(storageDirectory, "schedule.txt");
+        //createFileInDirectory(storageDirectory, "timetable.txt");
+    }
+
+
+
+    public Schedule loadSchedule() throws MissingFileException, CorruptedFileException {
```

Comment:
Excellent function, with clear logical progression and commenting which prevents the user from tripping up when reading the code. A five year old could read loadSchedule and understand. Nice work